app/src/ereferrals/java/org/eyeseetea/malariacare/domain/boundary/repositories/ICSVVersionRepository.java

```diff
@@ -0,0 +1,9 @@
+package org.eyeseetea.malariacare.domain.boundary.repositories;
+
+import org.eyeseetea.malariacare.data.IDataSourceCallback;
+
+public interface ICSVVersionRepository {
```

Comment:
Repositories and Datasources have the goal to the  domain abstraction of data layer and get and save domain entities but download csv process is not invoked from domain and is not entities, then DataSources and repositories concepts are wrong in this process, create interfaces for abstract is not necessary in this case because generate an unnecessary noise and complexity.